### DISCUSSION TITLE
Disconnect and later reconnect a dm object from a remote database

### DISCUSSION THREAD

**Discussion by ablack3** [cynkra/dm#1372] [General]
Would it be possible to disconnect and later reconnect a dm object to/from a remote database?

I'd like to pass a disconnected dm object into a function, have it open a new connection and perform some operations with the database, and then close the connection before exiting. One potential reason might be that on some database systems temporary tables are scoped to the connection which I want to keep within a function scope.

**Comment by krlmlr** [comment]
Thanks for raising this.

Have you seen `dm_paste()`? This prints the code that defines a dm.

There are a few open issues related to `dm_paste()`, would fixing any of those issues help? Please link to those issues from here and upvote them with a 👍  in the original post.
